src/geometry/adjoints.jl

```diff
@@ -49,3 +49,50 @@ function ChainRulesCore.rrule(::Type{SVector{N, T}}, x...) where {N, T}
     end
     return y, svector_const_pb
 end
+
+function ChainRulesCore.rrule(::typeof(ImageTransformations._getindex), A::AbstractExtrapolation, x)
+    y = A(Tuple(x)...)
+    function _getindex_pb(Δy)
+        # Δy :: NamedTuple{(:r, :g, :b)} or something similar
+        Δy = eltype(A)(Δy...)
+        gr = Interpolations.gradient(A, Tuple(x)...)
+        n = nfields(Δy) > 0 ? nfields(Δy) : 1
```

Comment:
this needs to be a method call that handles NamedTuples and Arrays separately. You need methods such as `_getindex_pb(Δy::NamedTuple{r,g,b})` and `_getindex_pb(Δy::AbstractArray)`

Reply:
We will never get arrays for `Δy` here. The reason is that we will be taking the gradients index-wise, so `Δy` will actually be always a `NamedTuple{...}`. I will only remove the broadcasted dot product with `n` and replace it with `*`.